meson.build

```diff
@@ -0,0 +1,484 @@
+project(
+    'libxkbcommon',
+    'c',
+    version: '0.7.1',
+    default_options: [
+        'c_std=c99',
+        'warning_level=2',
+        'b_lundef=true',
+    ],
+)
+pkgconfig = import('pkgconfig')
+cc = meson.get_compiler('c')
+
+
+# Compiler flags.
+foreach cflag: [
+    '-fvisibility=hidden',
+    '-Wextra',
+    '-Wno-unused-parameter',
+    '-Wno-missing-field-initializers',
+    '-Wpointer-arith',
+    '-Wmissing-declarations',
+    '-Wformat=2',
+    '-Wstrict-prototypes',
+    '-Wmissing-prototypes',
+    '-Wnested-externs',
+    '-Wbad-function-cast',
+    '-Wshadow',
+    '-Wlogical-op',
+    '-Wdate-time',
+    '-Wwrite-strings',
+]
+    if cc.has_argument(cflag)
+        add_project_arguments(cflag, language: 'c')
+    endif
+endforeach
+
+
+# The XKB config root.
+XKBCONFIGROOT = get_option('xkb-config-root')
+if XKBCONFIGROOT == ''
+    xkeyboard_config_dep = dependency('xkeyboard-config', required: false)
+    if xkeyboard_config_dep.found()
+        XKBCONFIGROOT = xkeyboard_config_dep.get_pkgconfig_variable('xkb_base')
+    else
+        XKBCONFIGROOT = join_paths(get_option('prefix'), get_option('datadir'), 'X11', 'xkb')
+  endif
+endif
+
+
+# The X locale directory for compose.
+XLOCALEDIR = get_option('x-locale-root')
+if XLOCALEDIR == ''
+    XLOCALEDIR = join_paths(get_option('prefix'), get_option('datadir'), 'X11', 'locale')
```

Comment:
I think I tried it originally (when adding compose). But I don't remember what was the problem.

Ideally libX11 would have a special variable for this path in its pc file, but it doesn't... So I'll try this.